src/canopy/llm/openai.py

```diff
@@ -206,3 +207,135 @@ async def agenerate_queries(self,
                                 model_params: Optional[dict] = None,
                                 ) -> List[Query]:
         raise NotImplementedError()
+
+
+class AzureOpenAILLM(OpenAILLM):
+    """
+    Azure OpenAI LLM wrapper built on top of the OpenAI Python client.
+
+    Note: OpenAI requires a valid API key to use this class.
+          You can set the "AZURE_OPENAI_KEY" environment variable to your API key.
+          Or you can directly set it as follows:
+          >>> import openai
+          >>> openai.api_key = "YOUR_API_KEY"
+    """
+    def __init__(self,
+                 model_name: str = "gpt-3.5-turbo",
+                 *,
+                 api_key: Optional[str] = None,
+                 base_url: Optional[str] = None,
+                 **kwargs: Any,
+                 ):
+        """
+        Initialize the Azure OpenAI LLM.
+
+        Args:
+            model_name: The name of the model to use. See https://platform.openai.com/docs/models
+            api_key: Your OpenAI API key. Defaults to None (uses the "OPENAI_API_KEY" environment variable).
+            organization: Your OpenAI organization. Defaults to None (uses the "OPENAI_ORG" environment variable if set, otherwise uses the "default" organization).
+            base_url: The base URL to use for the OpenAI API. Defaults to None (uses the default OpenAI API URL).
+            **kwargs: Generation default parameters to use for each request. See https://platform.openai.com/docs/api-reference/chat/create
+                    For example, you can set the temperature, top_p etc
+                    These params can be overridden by passing a `model_params` argument to the `chat_completion` or `enforced_function_call` methods.
+        """  # noqa: E501
+        super().__init__(model_name)
+        self._client = openai.AzureOpenAI(
+            api_key=api_key or os.getenv("OPENAI_API_KEY"),
```

Comment:
Note that the new openaiclient is automatically getting the AZURE_OPENAI_API_KEY param
https://github.com/openai/openai-python/blob/main/src/openai/lib/azure.py#L160